### ISSUE TITLE
Duckduino-microSD as Twin Duck

### ISSUE THREAD

**Issue by mapagm** [Seytonic/Duckduino-microSD#6]
Can it work as a Twin Duck? I mean that scenario: Arduino Pro Micro primarily acts as a USB Mass Storage Device, and on a click of the button will start emulating a Keyboard. Is it possible?

**Comment by draguve** [comment]
That shouldn't be a problem, but you would have to code a bit according to your use case

**Comment by WIFIDarthMaul** [comment]
I was wondering the same thing. I have this CJMCU Badusb board I bought on Ebay for about $10 and was wondering how to do it. The computer recognizes it as an Arduino Leonardo.
![a6rp_1_201711051437366327](https://user-images.githubusercontent.com/37938413/38147224-7ca659c0-340e-11e8-87ec-61d860197aaf.jpg)

**Comment by Eld0c** [comment]
@WIFIDarthMaul 
Did you know how to do it ?? cuz i'm gonna buy it . and need to know if it works or not ?
Do you recommend buy it or recommend diy it by myself ??

**Comment by WIFIDarthMaul** [comment]
@Eld0c Yeah it works good but I can't seem to do anything with the Twin Duck Firmware. But as a normal Ducky alternative, it works fine.

**Comment by Eld0c** [comment]
@WIFIDarthMaul 
is it work with this duckduino script ?
i wanna make it like ordinary flash memory , so i need once plugged in run my payload then works like storage . Is that possible ??

**Comment by ghost** [comment]
You can use this usb hub https://www.tindie.com/products/mux/nanohub-tiny-usb-hub-for-hacking-projects/ 
find small usb board and solder

**Comment by nibabaa** [comment]
yes,CJMCU can! i just succeed
script.txt 
[DELAY 3000
GUI r
DELAY 500
STRING notepad
DELAY 500
ENTER
DELAY 750
STRING Hello World!!!
ENTER]
and it run!
what i buy is CJMCU too
@mapagm

**Comment by nibabaa** [comment]
yes,
beacuse you can see, 
in the Duckduino-microSD.ino,
there is  ''    void Press(String b)
{   .....
 else if (b.equals("ENTER"))
  {
    Keyboard.press(KEY_RETURN);
  }
  else if (b.equals("CTRL"))
  {
    Keyboard.press(KEY_LEFT_CTRL);
  }
}
so u can use duck type code on it, it can transform

**Comment by nibabaa** [comment]
@mapagm

**Comment by WIFIDarthMaul** [comment]
What exactly does that code do? Turn the Duckuino MicroSD into a Twin Duck? @nibabaa

**Comment by john2ksonn** [comment]
You could code something yourself using  [LUFA](https://github.com/abcminiuser/lufa) or use it's examples.
